codex-rs/tui/src/onboarding/onboarding_screen.rs

```diff
@@ -337,6 +339,36 @@ pub(crate) async fn run_onboarding_app(
                     onboarding_screen.handle_paste(text);
                 }
                 TuiEvent::Draw => {
+                    if !did_full_clear_after_success
+                        && onboarding_screen.steps.iter().any(|step| {
+                            if let Step::Auth(w) = step {
+                                w.sign_in_state.read().is_ok_and(|g| {
+                                    matches!(&*g, super::auth::SignInState::ChatGptSuccessMessage)
+                                })
+                            } else {
+                                false
+                            }
+                        })
```

Comment:
So we iterate over `onboarding_screen.steps` and take the read lock of the mutex each time...should we just take the read lock and then iterate over `onboarding_screen.steps`?